OpenRA.Mods.Common/Traits/Render/WithIdleOverlay.cs

```diff
@@ -47,13 +48,13 @@ public IEnumerable<IActorPreview> RenderPreviewSprites(ActorPreviewInitializer i
 				p = init.WorldRenderer.Palette(Palette);
 
 			var body = init.Actor.TraitInfo<BodyOrientationInfo>();
-			var facing = init.Contains<FacingInit>() ? init.Get<FacingInit, int>() : 0;
-			var anim = new Animation(init.World, image, () => facing);
+			Func<int> facing = () => init.Contains<FacingInit>() ? init.Get<FacingInit, int>() : 0;
+			var anim = new Animation(init.World, image, facing);
 			anim.PlayRepeating(RenderSprites.NormalizeSequence(anim, init.GetDamageState(), Sequence));
 
-			var orientation = body.QuantizeOrientation(new WRot(WAngle.Zero, WAngle.Zero, WAngle.FromFacing(facing)), facings);
-			var offset = body.LocalToWorld(Offset.Rotate(orientation));
-			yield return new SpriteActorPreview(anim, offset, offset.Y + offset.Z + 1, p, rs.Scale);
+			Func<WRot> orientation = () => body.QuantizeOrientation(new WRot(WAngle.Zero, WAngle.Zero, WAngle.FromFacing(facing())), facings);
```

Comment:
You can simplify this down to `Func<WRot> orientation = () => body.QuantizeOrientation(WRot.FromFacing(facing()), facings);`